ldap/ldap_test.go

```diff
@@ -102,8 +148,23 @@ func (u *mockUser) Roles() []string     { return []string{} }
 
 var mockPutUser gimlet.User
 
+type putController int
+
+const (
+	putErr putController = iota
+	putSuccess
+)
+
+var putControl = putErr
+
 func mockPut(u gimlet.User) (string, error) {
```

Comment:
I'd have different mocks for each value of putControler and similar, and then just use different constructors as needed. 

(or a mockPutFactory that generates function objects as needed,) the global vars have a high-code-smell quotient, at least for me.

Reply:
If you don't want to get out of the globals, at least leave a comment describing the more ideal solution, to avoid a proliferation of this as an endorsed pattern :)

Reply:
Done